vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSearchController.java

```diff
@@ -0,0 +1,93 @@
+package edu.asu.diging.vspace.web.staff;
+
+import java.util.HashSet;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import edu.asu.diging.vspace.core.model.IVSpaceElement;
+import edu.asu.diging.vspace.core.services.IStaffSearchManager;
+
+@Controller
+public class StaffSearchController {
+
+    @Autowired
+    private IStaffSearchManager staffSearchManager;
+
+    @RequestMapping(value = "/staff/search")
+    public String searchInVspace(@RequestParam(value = "spacePagenum",required=false, defaultValue = "1") String spacePagenum, 
+            @RequestParam(value = "modulePagenum",required=false, defaultValue = "1") String modulePagenum,
+            @RequestParam(value = "slidePagenum",required=false, defaultValue = "1") String slidePagenum,
+            @RequestParam(value = "slideTextPagenum",required=false, defaultValue = "1") String slideTextPagenum,Model model,
+            @RequestParam(name = "searchText") String searchString, @RequestParam(value = "tab", defaultValue = "module") String tab) {
+
+        paginationForSpace(spacePagenum, model, searchString);
+
+        paginationForModule(modulePagenum, model, searchString);
+
+        paginationForSlide(slidePagenum, model, searchString);
+
+        paginationForSlideText(slideTextPagenum, model, searchString);
+        
+        model.addAttribute("searchWord", searchString);
+        model.addAttribute("activeTab", tab);
+        return "/staff/search/staffSearch";
+    }
+
+    /**
+     * This method is used for pagination of all searched string that belong to space table
+     * 
+     * @param spacePagenum ---> current page number sent as request parameter in the URL.
+     * @param model ---> This the object of Model attribute in spring spring MVC.
+     * @param searchString ---> This is the search string which is being searched.
+     */
+    private void paginationForSpace(String spacePagenum, Model model, String searchString) {
+        HashSet<IVSpaceElement> spaceSet = staffSearchManager.searchInSpaces(searchString,Integer.parseInt(spacePagenum));
```

Comment:
`searchInSpaces` should return a page object, so you don't have to make the same call again. for total pages just use the returned page object. same for the other methods that are similar.